examples/tutorials/streaming_api_tutorial.py

```diff
@@ -0,0 +1,954 @@
+"""
+Media Stream API
+================
+
+This tutorial shows how to use torchaudio's I/O stream API to
+fetch audio/video data, decode and apply preprocessings that
+libavfilter provides.
+
+"""
+
+######################################################################
+# 1. Overview
+# -----------
+#
+# Streaming API leverages the powerful I/O features of ffmpeg.
+#
+# It can
+#  - Load audio/video in variety of formats
+#  - Load audio/video from local/remote source
+#  - Load audio/video from microphone, camera and screen
+#  - Generate synthetic audio/video signals.
+#  - Load audio/video chunk by chunk
+#  - Change the sample rate / frame rate, image size, on-the-fly
+#  - Apply filters and preprocessings
+#
+# The streaming API works in three steps.
+#
+# 1. Opening media source (file, device, synthetic pattern generator)
+# 2. Configure output stream
+# 3. Stream the media
+#
+# At this moment, the features that the ffmpeg integration provide
+# are limited to the form of
+#
+# `<some media source> -> <optional processing> -> <tensor>`
+#
+# If you have other forms that can be useful to your usecases,
+# (such as integration with `torch.Tensor` type and file-like objects)
+# please file a feature request.
+#
+
+######################################################################
+# 2. Preparation
+# --------------
+#
+
+######################################################################
+#
+# .. note::
+#
+#    The streaming API requires FFmpeg libraries (>=4.1).
+#
+#    If you are using Anaconda Python distribution,
+#    ``conda install -c anaconda ffmpeg`` will install
+#    the required libraries.
+#
+#    When running this tutorial in Google Colab, the following
+#    command should do.
+#
+#    .. code::
+#
+#       !add-apt-repository -y ppa:savoury1/ffmpeg4
+#       !apt-get -qq install -y ffmpeg
+
+import IPython
+import matplotlib.pyplot as plt
+import torch
+import torchaudio
+from torchaudio.prototype.io import Streamer
+
+print(torch.__version__)
+print(torchaudio.__version__)
+
+base_url = "https://download.pytorch.org/torchaudio/tutorial-assets"
+AUDIO_URL = f"{base_url}/Lab41-SRI-VOiCES-src-sp0307-ch127535-sg0042.wav"
+VIDEO_URL = f"{base_url}/stream-api/NASAs_Most_Scientifically_Complex_Space_Observatory_Requires_Precision-MP4.mp4"
```

Comment:
What's the source of the video and audio files?

Reply:
audio is from VOiCES dataset (creative commons) and video is from nasa.gov (public domain). the license information is described in respective sections.